packages/flutter/test/gestures/long_press_test.dart

```diff
@@ -701,4 +701,45 @@ void main() {
     longPress.dispose();
     recognized.clear();
   });
+
+  testGesture('Switching buttons mid-stream does not fail to send "end" event', (GestureTester tester) {
+    final List<String> recognized = <String>[];
+    final LongPressGestureRecognizer longPress = LongPressGestureRecognizer()
+      ..onLongPressStart = (LongPressStartDetails details) {
+        recognized.add('primaryStart');
+      }
+      ..onLongPressEnd = (LongPressEndDetails details) {
+        recognized.add('primaryEnd');
+      };
+
+    const PointerDownEvent down4 = PointerDownEvent(
+      pointer: 8,
+      position: Offset(10, 10),
+    );
+
+    const PointerMoveEvent move4 = PointerMoveEvent(
+      pointer: 8,
+      position: Offset(100, 200),
+      buttons: kPrimaryButton | kSecondaryButton,
+    );
+
+    const PointerUpEvent up4 = PointerUpEvent(
+      pointer: 8,
+      position: Offset(100, 200),
+      buttons: kSecondaryButton,
```

Comment:
This might be outside of the scope of this PR, but when testing on macOS and web with a local example app the `buttons` of this final `PointerUpEvent` is actually `0`. Is this expected?